src/CommunityVoices/App/Website/View/Quote.php

```diff
@@ -350,25 +360,32 @@ public function getQuoteUpload($request, $errors = self::ERRORS_DEFAULT)
         $response = new HttpFoundation\Response($presentation->generate($domainXMLElement));
         $this->finalize($response);
         return $response;
+
     }
 
     public function postQuoteUpload($request, $errors = self::ERRORS_DEFAULT)
     {
-        if (!empty($errors->upload->errors)) {
+    // There are three possible outcomes when the user enteres a new quote:
+    // 1: Error with entering correct fields
+    // 2: Correct fields entered, user presses "batch submit"
+    // 3: Correct fields entered, user presses "submit"
+    // in cases 1 and 2, want to bring user back to same page with same fields filled in,
+    // in case 3, want to advance user to new screen showing their submitted quote
+        if (!empty($errors->upload->errors) or $request->request->get('batch_submit')) {
             return $this->getQuoteUpload($request, $errors);
         }
-
+        // if the user presses "submit"
         // We simply will show the edited quote.
         // dirname() removes the /new from the url we are
         // redirecting to.
-        $response = new HttpFoundation\RedirectResponse(
-            dirname($request->headers->get('referer')) . '/' . $errors->upload->quote->id[0]
-        );
-
-        $this->finalize($response);
-        return $response;
+        else {
```

Comment:
Please have the `else` right next to the end of the `if`, otherwise the control sequence will be really confusing.

Reply:
Okay. Should I place comments for else below?

Reply:
Yeah, within the else block is fine.